includes/apple-exporter/class-markdown.php

```diff
@@ -82,170 +62,190 @@ private function parse_nodes( $nodes ) {
 	/**
 	 * Parse an individual node for markdown.
 	 *
-	 * @param DomNode $node
-	 * @return string
+	 * @param DOMElement $node The node to process.
+	 *
 	 * @access private
+	 * @return string The processed content, converted to a Markdown string.
 	 */
-	private function parse_node( $node ) {
+	private function _parse_node( $node ) {
 		switch ( $node->nodeName ) {
-		case 'strong':
-			return $this->parse_strong_node( $node );
-		case  'i':
-		case 'em':
-			return $this->parse_emphasis_node( $node );
-		case 'br':
-			return $this->parse_linebreak_node( $node );
-		case 'p':
-			return $this->parse_paragraph_node( $node );
-		case 'a':
-			return $this->parse_hyperlink_node( $node );
-		case 'ul':
-			return $this->parse_unordered_list_node( $node );
-		case 'ol':
-			return $this->parse_ordered_list_node( $node );
-		case 'li':
-			return $this->parse_list_item_node( $node );
-		case 'h1':
-		case 'h2':
-		case 'h3':
-		case 'h4':
-		case 'h5':
-		case 'h6':
-			return $this->parse_heading_node( $node );
-		case '#comment':
-			return '';
-		case '#text':
-		default:
-			return $this->parse_text_node( $node );
+			case 'strong':
+				return $this->_parse_node_strong( $node );
+			case  'i':
+			case 'em':
+				return $this->_parse_node_emphasis( $node );
+			case 'br':
+				return '  ' . "\n";
+			case 'p':
+				return $this->_parse_node_paragraph( $node );
+			case 'a':
+				return $this->_parse_node_hyperlink( $node );
+			case 'ul':
+				return $this->_parse_node_unordered_list( $node );
+			case 'ol':
+				return $this->_parse_node_ordered_list( $node );
+			case 'li':
+				return $this->_parse_node_list_item( $node );
+			case 'h1':
+			case 'h2':
+			case 'h3':
+			case 'h4':
+			case 'h5':
+			case 'h6':
+				return $this->_parse_node_heading( $node );
+			case '#comment':
+				return '';
+			case '#text':
+			default:
+				return $this->_parse_node_text( $node );
 		}
 	}
 
 	/**
-	 * Convert a text node to markdown.
+	 * Convert an <em> node to markdown.
+	 *
+	 * @param DOMElement $node The node to process.
 	 *
-	 * @param DomNode $node
-	 * @return string
 	 * @access private
+	 * @return string The processed node, converted to a string.
 	 */
-	private function parse_text_node( $node ) {
-		return str_replace( '!', '\\!', $node->nodeValue );
+	private function _parse_node_emphasis( $node ) {
+		return '_' . $this->parse_nodes( $node->childNodes ) . '_';
 	}
 
 	/**
-	 * Convert a linebreak node to markdown.
+	 * Convert a heading node to markdown.
+	 *
+	 * @param DOMElement $node The node to process.
 	 *
-	 * @param DomNode $node
-	 * @return string
 	 * @access private
+	 * @return string The processed node, converted to a string.
 	 */
-	private function parse_linebreak_node( $node ) {
-		return "  \n";
+	private function _parse_node_heading( $node ) {
+
+		// Get the heading level.
+		preg_match( '/h(\d)/', $node->nodeName, $matches );
+
+		return sprintf(
+			'%s %s' . "\n",
+			str_repeat( '#', $matches[1] ),
```

Comment:
This point in the code is never reached unless an explicit match was already found, so `$matches[1]` will never be empty. However, your comment made me think about the fact that a `preg_match` is overkill for what we're doing here, so I can convert to a `substr`.